Remove usages of v1 Premises and Bedspace services

froddd opened the pull request:
# Context

https://dsdmoj.atlassian.net/browse/CAS-2081 (Epic: https://dsdmoj.atlassian.net/browse/CAS-1562)

# Changes in this PR

This PR prepares the codebase before moving to the `/cas3/v2` Premises and Bedspace API endpoints. 

- Replaces all uses of deprecated `Premises` and `TemporaryAccommodationPremises` types (and associated factories) with the newer `Cas3Premises`
- Removes unused non-namespaced premises and bedspace API endpoints
- Removes any references to `v2` to provide a clean base for the move to the 'new' v2: services, clients and utilities have been combined where possible, and any leftover in `v2` directories have been moved back to the main component hierarchy
- Removes several unused methods wherever they were found (I think most of these were in integration test pages)
- `turnaroundWorkingDayCount` used for the `PUT` premises endpoint has been renamed to `turnaroundWorkingDays` to match all other requests -- I think @muhammad-elabdulla will be pleased about that 😉 

Would also like to go further with the cleanup and remove a lot of utilities that are found in services (such as https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/blob/681239e28837224a34d8780bc8c09b43a9a3c97a/server/services/premisesService.ts#L165 and https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/blob/61ff818697cfb159bd24f82862b9517996669067/server/services/bedspaceService.ts#L195), as these really shouldn't appear there to keep concerns separated. However, this PR is becoming large enough as is, so I'll leave this for a later date!


## Screenshots of UI changes

No UI changes

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

<!-- Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible. You can monitor deployment failures in CircleCI
itself and application errors are found in
[Sentry](https://ministryofjustice.sentry.io/issues/?project=4504129156218880&referrer=sidebar&statsPeriod=24h).
Both events should be automatically sent to our [Slack
channel](https://mojdt.slack.com/archives/C048BJS7S2F). -->

froddd committed:
Remove usages of v1 Premises and Bedspace services

froddd committed:
Move mock apis around

This removes any references to V2

froddd committed:
Remove more unnecessary v2 mentions

froddd committed:
Reinstate New premises factory trait

froddd committed:
Remvoe further v2 directories

froddd committed:
Remove unused API endpoints

froddd committed:
Move v2 bedspace and premises utilities

froddd committed:
Retire deprecated turnaroundWorkingDayCount

gregkhawkins commented on server/@types/shared/models/Cas3UpdatePremises.ts:
```diff
@@ -13,6 +13,11 @@ export type Cas3UpdatePremises = {
+     * @deprecated
+     */
+    turnaroundWorkingDayCount?: number;
+    turnaroundWorkingDays?: number;
```
can probably merge this one in (to avoid this change in this PR):
https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/pull/1451/files#diff-a97de4ffa5ce9bbfbea71fc530cfd4886a58a7e16266a97d2c6d63fc349f8bfd

gregkhawkins commented on server/controllers/temporary-accommodation/manage/arrivalsController.ts:
```diff
@@ -30,7 +30,7 @@ export default class ArrivalsController {
       const callConfig = extractCallConfig(req)
 
-      const premises = await this.premisesService.getPremises(callConfig, premisesId)
+      const premises = await this.premisesService.getSinglePremises(callConfig, premisesId)
```
okay, so we are first integrating with the `GET /cas3/premises/{id}` here (and in all others areas of the application where we consume `GET /premises/{id}`). The idea being that we can get rid of the legacy gradually and in steps? This work was going to be done under this https://dsdmoj.atlassian.net/browse/CAS-2065 but we were thinking we would skip it to avoid conflicts with what you are working on and you go straight to `GET /cas3/v2/premises/{id}`.

That all said, just wanting to clarify my understanding here. It seems very sensible to go with this more iterative approach and to gradually get rid of the legacy and not take giant leaps.

froddd replied:
Yes, moving to the namespaced endpoints does 2 things: reduces the scope of old endpoints used (i.e. now all premises endpoints are from the `/cas3` namespace), and we remove the use of the 'old' `Premises` and `TemporaryAccomodationPremises` types, so that everything to do with premises uses the newer `Cas3Premises` one. Doing this first means it should be easier to then move to the `/cas3/v2` endpoints, as these all use the newer type. That's the theory, I do hope it's correct as the proof will be in the doing...

gregkhawkins replied:
all makes sense! Just wanted to check I was following the intended steps. Thanks!

gregkhawkins reviewed (commented):
Looking good - 1 comment / 1 question to clarify understanding

froddd committed:
Remvoe last few v2 directories

gregkhawkins reviewed (dismissed):
LGTM

AvraamK reviewed (dismissed):
All good to me!

SaintRhys commented on cypress_shared/pages/temporary-accommodation/manage/premisesList.ts:
```diff
@@ -1,7 +1,7 @@
 import type { TemporaryAccommodationPremises as Premises } from '@approved-premises/api'
```
There's a stray `TemporaryAccommodationPremises` here

SaintRhys reviewed (commented):
Do we want to shift the integration tests out of the `v2` folder - `integration_tests/tests/temporary-accommodation/manage/v2/bedspace.cy.ts`

and are we happy to keep all the audit events in `manage.ts` as V2s for the time being?

froddd commented:
@SaintRhys good catch on those integration tests -- I've just noticed them too and in the process of moving them.

froddd committed:
Remove more v2 directories for integration tests

froddd committed:
Remove more v2 directories in cypress shared

froddd committed:
Remove last few v2 playwright pages

froddd committed:
Move v2 views alongside existing

Update all import paths to views root, as this makes managing these
tempaltes easier.

froddd committed:
Fix Cypress E2E test issues

froddd committed:
Improve Cypress local config

This change prevents retries locally, but ensures DOM snapshots are
available for a number of tests to help with local debugging.

froddd committed:
Remove reference to V2 in clients

froddd commented:
All E2E tests passing locally:

```
$ npm run test:e2e

// ...

       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  apply-and-place.feature                  02:06        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  bedspaceSearch.feature                   00:20        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  booking-errors.feature                   01:48        6        6        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  booking.feature                          02:15        5        5        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  bookingSearch.feature                    00:47        2        2        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  lostBed.feature                          01:16        6        6        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        08:33       23       23        -        -        -  

$ npm run test:playwright:e2e:local

// ...

Running 20 tests using 2 workers
  20 passed (45.4s)

```
